python/lib/dependabot/python/file_updater/pipfile_preparer.rb

```diff
@@ -114,6 +114,24 @@ def pipfile_sources
             map { |h| h.dup.merge("url" => h["url"].gsub(%r{/*$}, "") + "/") }
         end
 
+        def sub_auth_url(source, credentials)
+          if source["url"].include?("${")
+            base_url = source["url"].sub(/\${.*}@/, "")
+
+            source_cred = credentials.
+                          select { |cred| cred["type"] == "python_index" }.
+                          find { |c| c["index-url"].sub(/\${.*}@/, "") == base_url }
+
+            return nil if source_cred.nil?
+
+            source["url"] = AuthedUrlBuilder.authed_url(credential: source_cred)
+            return source
+
```

Comment:
The above two lines (`return source` + blank line) are redundant.

Reply:
Thanks, I've removed them!